### PR TITLE
Use javax.servlet-api 3.1 for new flex project

### LINKED ISSUE #1674: Change the servlet api version for the sample Flex projects and fix at version 3.1
javax.servlet:javax.servlet-api

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1674.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -7,13 +7,15 @@ Bundle-SymbolicName: com.google.cloud.tools.eclipse.appengine.newproject.test
 Bundle-Vendor: Google Inc.
 Bundle-Version: 0.1.0.qualifier
 Fragment-Host: com.google.cloud.tools.eclipse.appengine.newproject
-Require-Bundle: org.junit;bundle-version="4.12.0",
+Require-Bundle: com.google.cloud.tools.eclipse.appengine.flex,
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/META-INF/MANIFEST.MF]: Import-Package is preferred these days for OSGI reasons. Could we use that instead?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProjectTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  private final IProgressMonitor monitor = new NullProgressMonitor();
+  private final AppEngineProjectConfig config = new AppEngineProjectConfig();
+  private IProject project;
+
+  @Before
+  public void setUp() {
+    IWorkspace workspace = ResourcesPlugin.getWorkspace();
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProjectTest.java]: or use @Rule TestProjectCreator?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java [hunk 3]
```diff
@@ -122,27 +121,26 @@ private void addDependenciesToProject(IProject project, IProgressMonitor monitor
...
         IFile destFile = libFolder.getFile(artifactFile.getName());
         destFile.create(new FileInputStream(artifactFile), true, subMonitor.newChild(30));
       } catch (CoreException ex) {
-        logger.log(Level.WARNING, "Error downloading " +
-            libraryFile.getMavenCoordinates().toString() + " from maven", ex);
+        logger.log(Level.WARNING, "Error downloading " + //$NON-NLS-1$
+            libraryFile.getMavenCoordinates().toString() + " from maven", ex); //$NON-NLS-1$
       } catch (FileNotFoundException ex) {
-        logger.log(Level.WARNING, "Error copying over " + artifactFile.toString() + " to " +
+        logger.log(Level.WARNING, "Error copying over " + artifactFile.toString() + " to " + //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java]: can artifactFile be null here?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProjectTest.java (hunk 2 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProjectTest.java]: I consulted the Standard counterpart, and it wasn't using `TestProjectCreator`. I wondered that too at first, but the reason may be that here we are in the process of creating a new project from scratch, while `TestProjectCreator` provides a fully working project.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/META-INF/MANIFEST.MF (hunk 1 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/META-INF/MANIFEST.MF]: This is the same problem with https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/1660#discussion_r108299807. This test attempts to install the flex facet, which requires the flex facet definition. However, the `.flex` bundle isn't in our target platform yet; this directive will pull in the bundle in testing.

When we eventually remove the `.flex` bundle, Eclipse will place an error marker in this file, so we won't forget removing this later.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java (hunk 3 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/CreateAppEngineFlexWtpProject.java]: The exception is from `new FileInputStream(artifactFile)`, so at that point `artifactFile` is not null. Also, `artifact.getFile()` never returns `null`, so we are good.

**Pull request merged**